novem/table/utils/format.py

```diff
@@ -86,3 +86,122 @@ def merge_from_index(src: Union[pd.DataFrame, pd.Index], io: Optional[int] = Non
             merge_instructions.append(f"{start_row + aio}:{row + aio} {level} lbl{len(merge_instructions) + 1}")
 
     return "\n".join(merge_instructions)
+
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None  # type: ignore
+
+
+def merge_from_index_first_rows(
+    src: Union[pd.DataFrame, pd.Index], io: Optional[int] = None, level: Optional[int] = None
+) -> str:
+    """
+    Get comma-separated list of first rows from each merge group at specified level.
+
+    Args:
+        src (Union[pd.DataFrame, pd.Index]): The source DataFrame or Index object to analyze.
+        io (Optional[int]): Initial offset. If provided, overrides the calculated offset.
+        level (Optional[int]): The level of the MultiIndex to analyze. If None, uses the
+            innermost (most granular) level.
+
+    Returns:
+        str: Comma-separated list of row numbers representing first rows of merged sections
+    """
+    if not isinstance(src, (pd.DataFrame, pd.Index)):
+        raise TypeError("Input must be a pandas DataFrame or Index object")
+
+    if isinstance(src, pd.DataFrame):
+        index = src.index
+        aio = src.columns.nlevels
+    else:
+        index = src
+        aio = 1
+
+    # Handle io parameter properly
+    offset = io if io is not None else aio
+
+    if not isinstance(index, pd.MultiIndex):
+        index = pd.MultiIndex.from_arrays([index])
+
+    if len(index) == 0:
+        return ""
+
+    # Handle level parameter properly
+    if level is None:
+        actual_level = index.nlevels - 1
+    else:
+        actual_level = level if level >= 0 else index.nlevels + level
+
+    if not 0 <= actual_level < index.nlevels:
+        raise ValueError(f"Level {level} out of range for index with {index.nlevels} levels")
+
+    first_rows: List[int] = []
+    current_label = None
+
+    for row, label in enumerate(index.get_level_values(actual_level)):
+        if label != current_label:
+            first_rows.append(row + offset)
+            current_label = label
+
+    return ",".join(map(str, first_rows))
+
+
+def merge_from_index_last_rows(
```

Comment:
I haven't look closely at this magic, but my gut tells me that `merge_from_index_first_rows` and `merge_from_index_last_rows` should have some code in common

Reply:
You're probably correct, but there are relatively few lines of code and the tests pass. Top and bottom selection also have just enough nuance that I don't feel like digging into the details.

Unless you have a very strong objection i'm inclined to let this DRY faux pas pass.

Reply:
> Oh, I've heard of "DRY", "**DO** Repeat Yourself", right!?

`YAGNI`, `DRY`, who comes up with these?

Joking aside, I've learned that I sometimes hug DRY a bit too closely too my chest. I'm trying to (re-)learn that copying code is OK. But in the end, it depends a lot on the level of similarity vs. specialization.

I took the liberty of diffing the two functions:

![image](https://github.com/user-attachments/assets/4c163801-c0a7-4adf-9c16-6bf9c416f27d)

Aside from the function name and docstring, these are literally the only differences and the diff is bigger than it seems simply due to variable naming. So, with an extra `if` statement and a `- 1` offset, I would probably de-dupe this myself.

Reply:
I've refactored as requested